python/libff/kv.py

```diff
@@ -155,35 +155,67 @@ class Shmm(kv):
     """ A local-like kv store. With python shared_memory package.
     Not allowed to modify the existing values."""
 
-    def __init__(self, size=4096, serialize=True):
+    def __init__(self, serialize=True):
+        """ Currently, value must be binary."""
         """ Currently, value must be binary."""
         self.serialize = serialize
-        self.size = size
-        self.map = {}   # key: name; value: (offset, number of bytes)
-        self.shm = shared_memory.SharedMemory(create=True, size=size)
-        self.offset = 0
+        self.offset = 10
+        #self.map = {} or, we can serialize it and put it in the shmm
+        # key: name; value: (offset, number of bytes)
+        # In context manager, start posix_ipc and terminate
+        # [10 offset, 10 key, 10 size, val, 10 key, 10 size, val, ...]
```

Comment:
why 10 bytes? I think you could just store an int64 (8 bytes) for offsets and sizes. I guess the key can be whatever you want (10 might be a little short, but we'll deal with that when it becomes a problem). You might look at https://docs.python.org/3/library/struct.html